api/src/main/java/com/datatorrent/api/DAG.java

```diff
@@ -281,6 +306,20 @@
   public abstract OperatorMeta getMeta(Operator operator);
 
   /**
+   * Return all operators present in the DAG.
+   * @param <T>
+   * @return
+   */
+  public <T extends OperatorMeta> Collection<T> getAllOperators();
+
+  /**
+   * Returns all Streams present in the DAG.
+   * @param <T>
+   * @return
+   */
+  public <T extends StreamMeta> Collection<T> getAllStreams();
```

Comment:
How about adding a similar methods to OperatorMeta to get list of input and output ports.